### ISSUE TITLE
[Feaure]: Access Permissions for android.permission.INTERNET

### ISSUE THREAD

**Issue by ColoursofOSINT** [celzero/rethink-app#1075]
Currently, it appears that RethinkDNS assumes that all apps have the android.permission.INTERNET, which as the name suggests grants internet access. 

However, many apps don't ask for this permission, and with custom ROMs, the permission can be revoked. This leads to a situation where an app has no internet access by the system Firewall (which can not be configured per port/ip ...etc, and not even configured with Stock ASOP) being shown as have access in Rethink.

This is confusing. So some type of permissions checking would fix this. Open-source Apps like [Permissions Pilot](https://github.com/d4rken-org/permission-pilot) can show these rules. As such, a feature where RethinkDNS checks permissions for android.permission.INTERNET, then updates the RethinkDNS firewall to shown that the system is blocking would be great.

Currently to find firewall state, one must check:

System Firewall then RethinkDNS firewall. 

With this feature, only RethinkDNS firewall would need to be checked.

**Comment by ColoursofOSINT** [comment]
Looks like this needs  QUERY_ALL_PACKAGES permission.

**Comment by ignoramous** [comment]
Makes sense, thanks. We'll see how we can present this information. Looks like its a useful indicator.

**Comment by ignoramous** [comment]
Starting `v055o`, Rethink grays out apps that *do not* request `android.permission.INTERNET` in *Configure -> Apps*.

Rethink still lets users setup firewall rules because an app may request for this permission after an update (and it'll be granted automatically at install time without a user prompt on most Android flavours other than GrapheneOS).
